examples/erp/genseki/collections/posts.tsx

```diff
@@ -167,6 +185,72 @@ const list = builder.list(fields, {
     sortBy: ['updatedAt', 'title'],
   },
   options: options,
+  actions: { create: true, delete: true, one: true, update: true },
+  uis: {
+    layout(args) {
+      const CollectionLayout = args.CollectionLayout
+      const CollectionSidebar = args.CollectionSidebar
+      const SidebarProvider = args.SidebarProvider
+      const SidebarInset = args.SidebarInset
+      const TopbarNav = args.TopbarNav
+
+      /**
+       * @description You can use following template for simple scaffolding
+       *  return (
+       *   <CollectionLayout>
+       *     <TopbarNav />
+       *     {args.children}
+       *   </CollectionLayout>
+       *  )
+       */
+
+      return (
+        <>
+          <SidebarProvider>
+            <CollectionSidebar />
+            <SidebarInset>
+              <TopbarNav />
+
+              <div>{args.children}</div>
+            </SidebarInset>
+          </SidebarProvider>
+        </>
+      )
+    },
+    pages(args) {
+      const clientListViewProps = getClientListViewProps(args.listViewProps)
+      const ListViewWrapper = args.ListViewWrapper
+      const ListView = args.ListView
+      const Banner = args.Banner
+
+      /**
+       * @description You can use following template for simple scaffolding
+       *
+       * ```return (
+       * <div>
+       *   <Banner />
+       *   <ListViewWrapper>
+       *     <ListView />
+       *   </ListViewWrapper>
+       * </div>
+       * )```
+       */
+
+      return (
+        <div>
+          <Banner />
+          <ListViewWrapper>
+            <TableStatesProvider>
+              <ClientListViewPropsProvider clientListViewProps={clientListViewProps}>
```

Comment:
Merge these two contexts into one. Since it always required 2 contexts.

```
`CollectionListProvider`
  - <TableStatesProvider>
  - <CollectionListProps />
```

Reply:
The expected convention of this should be

```tsx
<CollectionListProvider>
  <CollectionListBanner />
  <CollectionListToolbar />
  <CollectionListTable />
  <CollectionListPagination />
</CollectionListProvider>
```

Reply:
Maybe users does not need to insert `CollectionListPovider ` by themselve. We can hiddenly add the provider in `builder.list()`

As a result, user can have context to use but did not insert by themself.